### PR TITLE
Add AKSDockerEngine default for sovereign clouds

### LINKED ISSUE #114: Add AKSDockerEngine handler for sovereign clouds
AKSDockerEngine introduced in https://github.com/Azure/acs-engine/pull/4120 does not work for sovereignclouds, the value is always nil.

For example, when 'AKSDockerEngine' distro is selected, and resulting image publisher/offer would be empty. 

Error message:
```
{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.","details":[{"code":"BadRequest","message":"{\r\n \"error\": {\r\n \"code\": \"InvalidParameter\",\r\n \"message\": \"The value of parameter imageReference.publisher is invalid.\",\r\n \"target\": \"imageReference.publisher\"\r\n }\r\n}"}]}
```

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Fix missing defauld cloud spec distro for AKSDockerEngine distro. https://github.com/Azure/acs-engine/issues/4347

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #114 

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/azenvtypes.go [hunk 1]
```diff
@@ -244,14 +246,15 @@ var (
...
+			Ubuntu:          SovereignCloudsUbuntuImageConfig,
+			RHEL:            DefaultRHELOSImageConfig,
+			CoreOS:          DefaultCoreOSImageConfig,
+			AKS:             SovereignCloudsUbuntuImageConfig,
+			AKSDockerEngine: SovereignCloudsUbuntuImageConfig,
 		},
 	}
 
-	// AzureCloudSpecEnvMap is the environment configuration map for all the Azure cloid environments.
+	// AzureCloudSpecEnvMap is the environment configuration map for all the Azure cloud environments
```
Comment (mboersma) [inline pkg/api/azenvtypes.go]: Super nit-pick--can we leave the period intact? Go comments are meant to be complete sentences.

Comment (CecileRobertMichon) [reply pkg/api/azenvtypes.go]: whoops for sure

Comment (CecileRobertMichon) [reply pkg/api/azenvtypes.go]: added back

**Pull request merged**